tests/cloner_test.go

```diff
@@ -531,6 +531,65 @@ var _ = Describe("all clone tests", func() {
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				//TODO:  get storageClass `cloneStorageClassName` and annotate with `cloneType`
+				storageProfile.Spec.CloneStrategy = &cloneType
+
+			})
+			AfterEach(func() {
+				By("[AfterEach] Restore the storage class - remove annotation ")
+				// TODO: Remove annotation from storageClass `cloneStorageClassName`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				storageProfile.Spec.CloneStrategy = nil
+				storageProfile.Status.CloneStrategy = nil
+				_, err = f.CdiClient.CdiV1beta1().StorageProfiles().Update(context.TODO(), storageProfile, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				// TODO: verify that the storageProfile `cloneStorageClassName` has empty storageProfile.Status.CloneStrategy
+				storageProfile, err = f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+			})
+			It("Should clone  with correct strategy from storageclass annotation ", func() {
+				//TODO: get storageProfile and verify taht the cloneStrategy is set to `cloneType`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).ToNot(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).ToNot(BeNil())
+				Expect(*storageProfile.Spec.CloneStrategy).To(Equal(cloneType))
+				Expect(*storageProfile.Status.CloneStrategy).To(Equal(cloneType))
+
+				pvcDef := utils.NewPVCDefinition(sourcePVCName, "1Gi", nil, nil)
+				pvcDef.Spec.StorageClassName = &cloneStorageClassName
+				pvcDef.Namespace = f.Namespace.Name
+
+				sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile+"; chmod 660 "+testBaseDir+testFile)
+				doFileBasedCloneTest(f, pvcDef, f.Namespace, "target-dv", "1Gi")
+
+				targetDV := utils.NewCloningDataVolume("target-pvc", "1Gi", sourcePvc)
+
+				dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDV)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(utils.GetCloneType(f.CdiClient, dataVolume)).To(Equal(cloneType))
```

Comment:
That should be good

Reply:
actually it should be 
				Expect(utils.GetCloneType(f.CdiClient, dataVolume)).To(Equal("csivolumeclone"))